### ISSUE TITLE
TailCall isn't actually a TailCall

### ISSUE THREAD

**Issue by pdjonov** [henchmeninteractive/HenchLua#1]
This is tied to the fact that we currently create a stack frame for all Lua calls. Ideally we can avoid the overhead when making unprotected calls from bytecode to bytecode. This is blocked by the present lack of exception handling in the VM itself.

**Comment by pdjonov** [comment]
Calling this low priority since it doesn't affect correctness in nearly all cases.
